internal/pkg/cli/env_upgrade.go

```diff
@@ -37,12 +47,16 @@ type envUpgradeVars struct {
 type envUpgradeOpts struct {
 	envUpgradeVars
 
-	store environmentStore
-	sel   appEnvSelector
+	store       store
+	sel         appEnvSelector
+	envTemplate templater
+	prog        progress
 
 	// Constructors for clients that can be initialized only at runtime.
 	// These functions are overriden in tests to provide mocks.
-	newEnvVersionGetter func(app, env string) (versionGetter, error)
+	newEnvVersionGetter    func(app, env string) (versionGetter, error)
+	newTemplateUpgrader    func(conf *config.Environment) (envTemplateUpgrader, error)
+	newEnvOutputsDescriber func(app, env string) (envVPCDescriber, error)
```

Comment:
Do we use this anywhere 🤔

Reply:
🤦 lol i used it while experimenting if there was a way of reading the VPC configuration by checking the outputs of the cloudformation stack, and then forgot to remove the code.

Thanks for pointing it out!